beacon_node/beacon_chain/src/beacon_chain.rs

```diff
@@ -2616,21 +2590,29 @@ impl<T: BeaconChainTypes> BeaconChain<T> {
                 .map_err(BeaconChainError::from)?;
         }
 
-        let mut fork_choice = self.fork_choice.write();
+        let mut fork_choice = self.canonical_head.fork_choice_write_lock();
```

Comment:
```suggestion
```

Reply:
I don't quite understand this suggestion. I added some comments to this call though, warning of the perils of deadlocks.

Reply:
Oh I think I understand now, with the context of the next comment!